app/controllers/katello/api/v2/subscriptions_controller.rb

```diff
@@ -77,8 +77,8 @@ def index_relation
     def show
       org_id = Organization.current&.id
       @resource = Katello::Pool.with_identifier(params[:id])
-      if @resource.organization_id != org_id
-        fail ActiveRecord::RecordNotFound, N_('This subscription is not relevant to the current organization.')
+      if @resource.organization_id != org_id && !User.current.organizations&.pluck(:id)&.include?(@resource.organization_id)
+        fail ActiveRecord::RecordNotFound, N_('This subscription is not relevant to the current user and organization.')
```

Comment:
When I give the API a bogus subscription ID I get a stack trace. Might be worth addressing here as well:

```
curl -X GET -k -u qauser:changeme http://localhost:3000/katello/api/v2/organizations/4/subscriptions/5000
{"displayMessage":"undefined method `organization_id' for nil:NilClass","errors":["undefined method `organization_id' for nil:NilClass"]}```